gui/src-tauri/src/main.rs

```diff
@@ -159,19 +196,29 @@ fn main() -> Result<()> {
 			.then(|| jre.into_os_string())
 			.or_else(|| valid_java_paths().first().map(|x| x.0.to_owned()));
 		let Some(java_bin) = java_bin else {
-			show_error(&format!("Couldn't find a compatible Java version, please download Java {} or higher", MINIMUM_JAVA_VERSION));
-			return Ok(());
+			show_error(&format!(
```

Comment:
use `formatcp!` instead as it can be done constant-time

Reply:
Uh, you know formatcp! is a compile-time thing, right? But since we use a runtime variable, that won't exactly... work.

Reply:
```
the trait bound `std::string::String: From<&&str>` is not satisfied
required for `&&str` to implement `Into<std::string::String>`

attempt to use a non-constant value in a constant
```

Reply:
its because you doing `&formatcp!` instead of just `formatcp!`

Reply:
This is what I get for waking up 20 mins ago

Reply:
@ImUrX That... Still gets you `attempt to use a non-constant value in a constant` because of `checked_envs`

Reply:
only use formatcp on formats with only constant values